plugins/statuslines/lightline.nix

```diff
@@ -1,117 +1,209 @@
 {
   lib,
-  helpers,
   config,
+  options,
   pkgs,
   ...
 }:
-with lib;
 let
-  cfg = config.plugins.lightline;
+  inherit (lib) types;
+  inherit (lib.nixvim) defaultNullOpts;
 in
-{
-  options = {
-    plugins.lightline = {
-      enable = mkEnableOption "lightline";
+lib.nixvim.neovim-plugin.mkNeovimPlugin config {
+  name = "lightline";
+  originalName = "lightline.vim";
+  defaultPackage = pkgs.vimPlugins.lightline-vim;
 
-      package = helpers.mkPluginPackageOption "lightline" pkgs.vimPlugins.lightline-vim;
+  maintainers = [ lib.maintainers.khaneliman ];
 
-      colorscheme = mkOption {
-        type = with types; nullOr str;
-        default = config.colorscheme;
-        description = "The colorscheme to use for lightline. Defaults to .colorscheme.";
-        example = "gruvbox";
-      };
+  # TODO: Added 2024-08-23, remove after 24.11
+  optionsRenamedToSettings = [
+    "colorscheme"
+    "componentFunction"
+    "component"
+    "active"
+    "inactive"
+    "modeMap"
+  ];
 
-      componentFunction = mkOption {
-        default = null;
-        type = with types; nullOr (attrsOf str);
-        description = ''
-          A list of function component definitions.
+  settingsOptions = {
+    colorscheme = defaultNullOpts.mkStr "default" ''
+      The colorscheme to use for lightline.
+      Default theme is equal to `powerline`.
+    '';
 
-          You should define the functions themselves in extraConfig
-        '';
-        example = ''
-          plugins.lightline = {
-            enable = true;
-            componentFunction = {
-              readonly = "LightlineReadonly";
-            };
+    component_function = defaultNullOpts.mkAttrsOf' {
+      type = types.str;
+      pluginDefault = { };
+      description = ''
+        A list of function component definitions.
 
-            extraConfig = '''
-              function! LightlineReadonly()
-                return &readonly && &filetype !=# 'help' ? 'RO' : '''
-              endfunction
-            ''';
+        You can use the name of a function defined elsewhere.
+      '';
+      example = ''
+        plugins.lightline = {
+          enable = true;
+          settings.component_function = {
+            readonly = "LightlineReadonly";
           };
+        };
```

Comment:
This isn't an option example; option examples are usually _just_ the value of the option. In this case:
```nix
{
  readonly = "LightlineReadonly";
}
```

Reply:
I guess if the purpose of the example is to demonstrate referencing a function defined elsewhere, perhaps that should be documented in the option's description?

Or at least make an example using `literalMD` or `literalExpression`.